### DISCUSSION TITLE
Am I doing this basic auth wrong? works in developments, fails when deployed

### DISCUSSION THREAD

**Discussion by Konstantin-Koktashev** [nextauthjs/next-auth#2315] [Help]
### Question 💬

hey guys, Im using vercel to host,
Google auth works fine locally but once hosted, first click causes refresh, second one causes this error:
```
Error 400: invalid_request
Missing required parameter: client_id
```
I've set my client id both in vercel Environment Variables
and in my env.local 

```
import NextAuth from "next-auth";
import Providers from "next-auth/providers";
export default NextAuth({
  // Configure one or more authentication providers
  providers: [
    Providers.Google({
      clientId: process.env.GOOGLE_ID,
      clientSecret: process.env.GOOGLE_SECRET,
    }),
    // ...add more providers here
  ],
});
```

### How to reproduce ☕️

```
import {signIn,signOut,useSession} from "next-auth/client";
const [session]=useSession();
        <div onClick={!session ? signIn : signOut} >
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by mahieyin-rahmun** [comment]
> Error 400: invalid_request
> Missing required parameter: client_id

This usually means it cannot load the variable. Check if you have a key mismatch e.g. you typed something other than ` GOOGLE_ID` in your environment variables setting dashboard at Vercel.

Also, check if you set the NEXTAUTH_URL properly.

**Comment by Konstantin-Koktashev** [comment]
I've checked again,NEXTAUTH_URL is set to the website address itself provided by vercel,
GOOGLE_ID is the same as on the dev environment,
after the first click my adress bar changes from this :

/api/auth/signin?callbackUrl=https%3A%2F%2Famazon-clone-lclkx7dr0-koktashev33-gmailcom.vercel.app%2F

to this: 

api/auth/signin?csrf=true 

and the page refreshes, on the second click i get the error, could it have anything to do with it?
here is the repo if that helps : https://github.com/Konstantin-Koktashev/amazon-clone

**Reply by balazsorban44** [reply]
⚠️⚠️⚠️⚠️

@Konstantin-Koktashev you should immediately set the repo to private, and change your permissions, as I'm pretty sure you exposed private keys in that repo!

you will need to regenerate them in Google, don't use the ones currently exposed.

**Reply by Konstantin-Koktashev** [reply]
Thank you, forgot about that detail I'm a novice , will fix it now.
Did u manage to see by any chance the code or have any idea what could be the problem?

**Reply by 0ubbe** [reply]
If it's working locally but not in **Vercel**, prolly is a misconfiguration with the [environment variable in **Vercel**](https://vercel.com/docs/environment-variables) 🙇🏽‍♂️ and not with the library itself, not sure what else can we do to help you.
